### DISCUSSION TITLE
RFC: Improvements to console tracking UI

### DISCUSSION THREAD

**Discussion by vijayprasanna13** [hasura/graphql-engine#9727] [General] [answered]
Hey folks, we're making improvements to console UI surrounding the tracking section for databases. As a first step, we're making an improved version of the "Table Tracking" section that should be available with an opt-in feature flag on the console. 

![Screenshot 2023-06-21 at 11 44 30 AM](https://github.com/hasura/graphql-engine/assets/11921040/23a29fed-7cbd-42d2-baa8-45cadc9a01d5)

The features that are coming along with this are - 
- Pagination for the list of tables
- A filter-by table type option
- A dedicated section to see the list of previously tracked tables
- Text search based on the table name

We'll keep this discussion updated on the progress and when it's coming out. Please do give it a try when it does and let us know your feedback or catch any bugs! Thanks!

**Comment by vijayprasanna13** [comment]
Hey folks! You can try out the feature early by using the demo image below 
```bash
docker pull hasura/graphql-engine-demo:mono-pr-9578-legacy.ubuntu
```

**How to enable this feature?**
Navigate to the `Settings > Feature Flags` part of the console and toggle the `Enable new Table Tracking UI for Postgres & SQL Server` feature flag to start using this feature


https://github.com/hasura/graphql-engine/assets/11921040/9a0603aa-04ce-4efb-89df-3b1185fc35fb

**Reply by jjstanton** [reply]
Thank you for this! Unfortunately, I've encountered an fetch error for the assetLoader.js.gz.

Would you be able to confirm this?

![image](https://github.com/hasura/graphql-engine/assets/53913264/1a5ae37c-414e-459c-9cfa-050be24c37b9)

**Reply by vijayprasanna13** [reply]
this might be a CDN issue since the older assets could have removed. I'd suggest trying out `v2.29.0` version of Hasura!

**Comment by anish-pr** [comment]
We also got the same issue, the ui is not loading with this image. Stuck in loading screen

**Comment by vijayprasanna13** [comment]
@anish-pr @jjstanton Could try the `v2.29.0` version of Hasura?

**Reply by jjstanton** [reply]
Thanks, v2.29.0 does indeed solve the issue. Enjoying the new features!

**Comment by vijayprasanna13** [comment, marked as answer]
Hey folks! We have more updates! From `v2.30.0` onwards the new and improved tracking functionality is now available for suggested relationships. It brings -
1. Search and Pagination
2. Way faster tracking and un-tracking of multiple relationships at once.
3. Ability Customize you suggested relationship name before tracking them.
![track_relationships](https://github.com/hasura/graphql-engine/assets/11921040/bb5539d3-15b3-4d9a-8b47-041be5dac226)

**Comment by thinknirmal** [comment]
After adding a new foreign key to an existing table, the new tracking UI doesn't suggest the untracked relationships. The old UI still correctly suggests them.

<img width="477" alt="image" src="https://github.com/hasura/graphql-engine/assets/350161/a7a09538-bb91-4f08-ae0d-c55b2e3b03ae">
<img width="503" alt="image" src="https://github.com/hasura/graphql-engine/assets/350161/40c0f159-c438-456f-9863-e80379196c6d">

This doesn't always happen. It's more of 50/50 chances of success.

**Reply by vijayprasanna13** [reply]
Thanks @thinknirmal for the feedback! we'll look into this!

**Reply by vijayprasanna13** [reply]
> This doesn't always happen. It's more of 50/50 chances of success.

The explanation for this part is that we don't use raw SQL anymore to fetch the Foreign key info. This was done for performance reasons. As a solution, we could have a `refresh` button near the list to fetch the latest info schema info in case.

**Comment by IS-Josh** [comment]
We use Alloydb with partitioned tables.  The Data page in Hasura console is still unusable. Querires time-out.

**Comment by frasmarco** [comment]
The new UI is so usefull! Is it possible to have pagination settings saved? Actually when coming back the number of item per page resets to 10.
Many thanks
